.changeset/tasty-news-change.md

```diff
@@ -0,0 +1,18 @@
+---
+"@comet/cms-api": minor
+---
+
+Add `ExceptionFilter` to replace `ExceptionInterceptor`
+
+The main motivation for this change was that the `ExceptionInterceptor` didn't capture exceptions thrown in guards. This could lead to information leaks, e.g. details about the database schema or the underlying code. This is considered a security risk.
+
+The `ExceptionFilter` also catches error within guards. The error format remains unchanged.
+
+Switching from the `ExceptionInterceptor` to the `ExceptionFilter` must be done in the project:
+
+```diff
+// main.ts
+
+- app.useGlobalInterceptors(new ExceptionInterceptor(config.debug));
++ app.useGlobalFilters(new ExceptionFilter(config.debug));
```

Comment:
Should we add an upgrade script for this?

Reply:
https://github.com/vivid-planet/comet-upgrade/pull/46